ErrnoException

In version 2.0.0

```
java.lang.AssertionError: libcore.io.ErrnoException: getsockname failed: EBADF (Bad file number)
            at java.nio.SocketChannelImpl.initLocalAddressAndPort(SocketChannelImpl.java:217)
            at java.nio.SocketChannelImpl.connect(SocketChannelImpl.java:195)
            at com.koushikdutta.async.AsyncServer$6.run(AsyncServer.java:362)
            at com.koushikdutta.async.AsyncServer.lockAndRunQueue(AsyncServer.java:708)
            at com.koushikdutta.async.AsyncServer.runLoop(AsyncServer.java:725)
            at com.koushikdutta.async.AsyncServer.run(AsyncServer.java:626)
            at com.koushikdutta.async.AsyncServer.access$700(AsyncServer.java:41)
            at com.koushikdutta.async.AsyncServer$13.run(AsyncServer.java:568)
     Caused by: libcore.io.ErrnoException: getsockname failed: EBADF (Bad file number)
            at libcore.io.Posix.getsockname(Native Method)
            at libcore.io.ForwardingOs.getsockname(ForwardingOs.java:72)
            at java.nio.SocketChannelImpl.initLocalAddressAndPort(SocketChannelImpl.java:215)
            at java.nio.SocketChannelImpl.connect(SocketChannelImpl.java:195)
            at com.koushikdutta.async.AsyncServer$6.run(AsyncServer.java:362)
            at com.koushikdutta.async.AsyncServer.lockAndRunQueue(AsyncServer.java:708)
            at com.koushikdutta.async.AsyncServer.runLoop(AsyncServer.java:725)
            at com.koushikdutta.async.AsyncServer.run(AsyncServer.java:626)
            at com.koushikdutta.async.AsyncServer.access$700(AsyncServer.java:41)
            at com.koushikdutta.async.AsyncServer$13.run(AsyncServer.java:568)
```

--- a/AndroidAsync/src/com/koushikdutta/async/AsyncServer.java
+++ b/AndroidAsync/src/com/koushikdutta/async/AsyncServer.java
@@ -361,11 +361,11 @@ public class AsyncServer {
                     ckey.attach(cancel);
                     socket.connect(address);
                 }
-                catch (IOException e) {
+                catch (Throwable e) {
                     if (ckey != null)
                         ckey.cancel();
                     StreamUtility.closeQuietly(socket);
-                    cancel.setComplete(e);
+                    cancel.setComplete(new RuntimeException(e));
                 }
             }
         });

